langserver/signature.go

```diff
@@ -22,20 +22,20 @@ import (
 )
 
 // SignatureHelp is required by the protocol.Server interface.
-func (s *server) SignatureHelp(ctx context.Context, params *protocol.SignatureHelpParams) (*protocol.SignatureHelp, error) {
+func (s *server) SignatureHelp(_ context.Context, params *protocol.SignatureHelpParams) (*protocol.SignatureHelp, error) {
 	location, err := s.cache.Find(&params.TextDocumentPositionParams)
 	if err != nil {
-		return nil, nil
+		return nil, err
 	}
 
 	call, ok := location.Node.(*promql.Call)
 	if !ok {
-		return nil, nil
+		return nil, errors.New("no node find for the given query")
```

Comment:
This code branch is not an error condition. In case we reach this, the cursor is not over a function call and we simply return an empty result.

Also note, that returning `nil` pointers isn't a problem with this endpoint, since no member accesses take place.